src/cpp/polars/TimeSeries.h

```diff
@@ -0,0 +1,143 @@
+//
+// Created by Linda Uruchurtu on 03/10/2018.
+//
+
+#ifndef POLARS_TIMESERIES_H
+#define POLARS_TIMESERIES_H
+
+#include <cassert>
+#include <vector>
+#include <cmath>
+#include <string>
+#include "armadillo"
+#include <chrono>
+#include "date/date.h"
+
+#include "Series.h"
+
+
+namespace polars {
+
+    using namespace std::chrono;
+    typedef std::chrono::duration<double> unix_epoch_seconds;
+
+    template<class TimePointType>
+    class TimeSeries : public Series {
+
+    public:
+        using Series::loc;
+
+        TimeSeries() = default;
+
+        TimeSeries(arma::vec v0, std::vector<TimePointType> t0){
+            v = v0;
+            t = chrono_to_double_vector(t0);
+        };
+
+        std::vector<TimePointType> timestamps() const {
+            // Pass indices and return vector of timepoints
+            return double_to_chrono_vector(index());
+        };
+
+        std::map<TimePointType, double> to_timeseries_map() const {
+
+            std::map<TimePointType, double> m;
+
+            std::vector<TimePointType> timepoints = double_to_chrono_vector(index());
+
+            // Put pairs into map
+            for(int i = 0; i < size(); i++) {
+                m.insert(std::make_pair(timepoints[i], values()[i]));
+            }
+
+            return m;
+        };
+
+        TimeSeries loc(const std::vector<TimePointType> &index_labels) const {
+            std::vector<double> indices;
+
+            // Turn time_points to doubles
+            auto tstamps = chrono_to_double_vector(index_labels);
+
+            for (int j = 0; j < tstamps.n_elem; j++) {
+
+                arma::uvec idx = arma::find(index() == tstamps[j]);
```

Comment:
This is a very expensive way to do this as it is O(n * m) where n is the index size and m is size of `index_labels`. Maybe add a TODO saying we should find a more efficient way to do this.

Reply:
The more I think about indexes, the more I think we should make them objects in their own right (like pandas does). We can then deal with the underlying implementation however we like and any improvements get shared. No need to do anything atm, but worth thinking about how we might make the transition in the future.

Reply:
Added todos